### ISSUE TITLE
Can't build master on VS2013 due to snprintf in core\system.cpp 

### ISSUE THREAD

**Issue by PkLab** [opencv/opencv#8768]
##### System information (version)
- OpenCV => master 0bc30d382d
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2013

##### Detailed description
After  #8596  (_Don't blow out the stack. Use a smaller buffer and prevent buffer overruns with snprintf_)
Building with above environment you will have 
`C3861: 'snprintf': identifier not foun` in [system.cpp](https://github.com/opencv/opencv/blob/0bc30d382d2813896cb2032533a0948a83e8bcf1/modules/core/src/system.cpp#L864cpp)

this is because 'snprintf' isn't available on VS2013 see [ticket 809403](https://connect.microsoft.com/VisualStudio/feedback/details/809403/error-c3861-snprintf-identifier-not-found-in-visual-studio-2013). A possible workaround is to use `#define snprintf _snprintf` even if [msdn says](https://msdn.microsoft.com/en-us/library/2ts7cx93.aspx)

> in Visual Studio 2015 and Windows 10, snprintf is no longer identical to _snprintf. 

please rollback a solution

**Comment by catree** [comment]
Maybe?

```
#if defined _MSC_VER && _MSC_VER <= 1800
    sprintf(...)
    //Or maybe?
    _snprintf(...)
#else
    snprintf(...)
#endif
```

**Comment by mislavcro** [comment]
I'm having the exact same problem. Tried on 3 computers, all from fresh start, and it's not possible to work.

**Comment by alalek** [comment]
MSVS2013 need something like this: https://github.com/google/glog/blob/77a0e36b7a543a534b2a3634f06c958a638a40a4/src/windows/port.cc#L51-L56

**Comment by mislavcro** [comment]
> alalek commented 8 minutes ago
MSVS2013 need something like this: https://github.com/google/glog/blob/77a0e36b7a543a534b2a3634f06c958a638a40a4/src/windows/port.cc#L51-L56

Why this? Could I just write that code in OpanCV and will it work?

**Comment by alalek** [comment]
Because behavior of `_snprintf` from MSVC is not equal to standard `snprintf` behaviour: https://msdn.microsoft.com/en-us/library/2ts7cx93.aspx

**Comment by mislavcro** [comment]
So is there a way to use in OpanCV for building? That code you post works here maybe?

**Comment by PkLab** [comment]
In addiction `snprintf` is since C++11.

I would point that

- we are in `void error( const Exception& exc )`.
- The PR reduces memory allocation for buf.
- Using snprintf, in case of over buffer, introduces truncation instead of empty string. Is there any other way to achieve same result 
- How many `sprintf` we have in OpenCV ? 

... in the meanwhile we can also undo the commit

**Comment by alalek** [comment]
@PkLab Could you please check #8771 ?

**Comment by PkLab** [comment]
It compile but I'm not sure this will works correctly.
check my comment  to the PR

**Comment by PkLab** [comment]
I think this can be closed because solved in  _fix snprintf() usage_ #8771 and _Fixed snprintf for VS 2013_ #8816
